### DISCUSSION TITLE
Transform to Concatenate Images to Multi-channel Image

### DISCUSSION THREAD

**Discussion by dyollb** [Project-MONAI/MONAI#4515] [Q&A] [answered]
I would like to train with different combinations of input image modalities (or masks). So far I have used e.g. nibabel or sitk to concatenate T1, T2, etc. to generate a single image file. Is there any way I can achieve this currently in monai? 

If this were a feature request, I guess I would like to do something like this:
```
transforms = Compose([
    LoadImaged(keys=["t1w", "t2w", "label"]),
    EnsureChannelFirstd(keys="t1w", "t2w", "label"]),
    ConcatenateChannelsd(keys=["t1w", "t2w"], new_key="image"),
    NormalizeIntensityd(keys=["image", "label"]),
])
```

Another option I could think of is to first concatenate all possible channels, and then add a skip channel transform (could not find something like that in the docs) after the `LoadImaged`, e.g.
```
transforms = Compose([
    LoadImaged(keys=["image", "label"]),
    SkipChannelsd(keys="image", channels=[1,4]),
    NormalizeIntensityd(keys=["image", "label"]),
])
```

Is there any existing solution for this use-case?

**Comment by Nic-Ma** [comment, marked as answer]
Hi @dyollb ,

I think maybe you can try this transform `ConcatItemsd`:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/utility/dictionary.py#L951

Thanks.

**Reply by dyollb** [reply]
this looks exactly like what I was looking for. thanks!
